src/util/secure_socket_thread.c

```diff
@@ -0,0 +1,211 @@
+#include <string.h>
+#include <pthread.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/epoll.h>
+#include <fcntl.h>
+#include <kmyth/formatting_tools.h>
+
+#include "charbuf.h"
+#include "pelz_log.h"
+#include "pelz_socket.h"
+#include "pelz_json_parser.h"
+#include "pelz_io.h"
+#include "pelz_service.h"
+#include "pelz_request_handler.h"
+#include "secure_socket_thread.h"
+
+#include "sgx_urts.h"
+#include "pelz_enclave.h"
+#include ENCLAVE_HEADER_UNTRUSTED
+
+#define BUFSIZE 1024
+#define MODE 0600
+
+static void *secure_process_wrapper(void *arg)
+{
+  secure_socket_process(arg);
+  pthread_exit(NULL);
+}
+
+void secure_socket_thread(void *arg)
+{
+  ThreadArgs *threadArgs = (ThreadArgs *) arg;
+  int socket_listen_id = threadArgs->socket_id;
+  int max_requests = threadArgs->max_requests;
+  pthread_mutex_t lock = threadArgs->lock;
+
+  ThreadArgs processArgs;
+  pthread_t stid[max_requests];
+  int socket_id = 0;
+
+  do
+  {
+    if (pelz_key_socket_accept(socket_listen_id, &socket_id))
+    {
+      pelz_log(LOG_ERR, "Socket Client Connection Error");
+      continue;
+    }
+
+    if (socket_id == 0)         //This is to reset the while loop if select() times out
+    {
+      continue;
+    }
+
+    if (socket_id > max_requests)
```

Comment:
I think socket_id is a file descriptor number here, not a count of the number of connections.
This also has implications for the channel table functions, which use `socket_id` as an array index.

Reply:
When I try it, the first socket_id is 6, not 0.